tests/php/Core/ClassInfoTest.php

```diff
@@ -189,4 +199,67 @@ public function testDataClassesFor()
         ClassInfo::reset_db_cache();
         $this->assertEquals($expect, ClassInfo::dataClassesFor(strtolower($classes[2])));
     }
+
+    /**
+     * @covers \SilverStripe\Core\ClassInfo::classesWithExtension()
+     */
```

Comment:
We don't really need this annotation, as you're calling that method directly within the test, so it's implied. We can use this annotation when the test covers similar logic in another place that isn't directly tested.

Reply:
My understanding is that this also stops code coverage from being tracked in other methods apart from those listed in `@covers`, so this is a good thing to add when you don't want to imply that you're testing other code that `classesWithExtension()` calls internally. Not sure if that's intentional or not, but IMO adding `@covers` where relevant helps distinguish code that's intentionally tested vs. code that's covered but not intentionally.

Reply:
Interesting point, I can't really see how that would work though - does it tell PHPUnit that it only covers the method you annotate and no other method in the call stack?

Reply:
I just followed the pattern for other test methods in the test class, so it's not a thing I added as a new concept.